pageserver/src/tenant/remote_timeline_client.rs

```diff
@@ -949,6 +949,35 @@ impl RemoteTimelineClient {
         Ok(())
     }
 
+    /// If the `import_pgdata` field marks the timeline as having an import in progress,
+    /// launch an index-file upload operation that transitions it to done in the background
+    pub(crate) fn schedule_index_upload_for_import_pgdata_finalize(
+        self: &Arc<Self>,
+    ) -> anyhow::Result<()> {
```

Comment:
This method should fail with an error if called on an import that is not done yet.
We must not allow transition of a timeline that is still in progress importing into Done state.
Should also make sure there's a comment explaining how load-bearing that check is.

Reply:
When you say "import is not done yet" - what do you mean?

This function will only be called after the import reported itself as done in storcon.
I agree with the overall point that the import job may still be running, but it wouldn't just exit immediately
after seeing that the storcon status for the shard is Done.

Reply:
This is about robustness: a bug in storcon that calls this function early would cause us to mark the timeline as done even though the import job is still running.

Reply:
The method itself doesn't have the right context to check if the import task is on-going, but I've added a check at the call site in: https://github.com/neondatabase/neon/commit/8fa36e45e734c0a9c7b504451d32ea6971235e11